gadgets/traceloop/README.mdx

```diff
@@ -146,6 +186,133 @@ Finally, clean the system:
     </TabItem>
 </Tabs>
 
+## Advanced Tracing Scenarios
+
+### Built-in System Call Filtering
+
+For more efficient filtering, traceloop supports built-in syscall filtering at the eBPF level, which reduces overhead and provides cleaner output than post-processing with grep.
+
+<Tabs groupId="env">
+    <TabItem value="kubectl-gadget" label="kubectl gadget">
+        ```bash
+        # Filter for specific syscalls
+        $ kubectl gadget run traceloop:latest -n test-ns --podname my-pod --syscall-filters openat,write,read
+        # Focus on file operations only
+        $ kubectl gadget run traceloop:latest -n test-ns --syscall-filters openat,close,read,write,unlink
+        ```
+    </TabItem>
+
+    <TabItem value="ig" label="ig">
+        ```bash
+        # Filter for specific syscalls (using -c shorthand)
+        $ sudo ig run traceloop:latest -c test-container --syscall-filters openat,write
+        # Focus on file operations only
+        $ sudo ig run traceloop:latest -c test-container --syscall-filters openat,close,read,write,unlink
+        ```
+    </TabItem>
+</Tabs>
+
+**Benefits of built-in filtering:**
+- **Reduced overhead**: Filtering happens at the kernel level, not in userspace
+- **Cleaner output**: Only relevant syscalls are captured and displayed
+- **Better performance**: Less CPU and memory usage compared to grep filtering
+- **Focused debugging**: Easier to spot patterns in specific syscall categories
+
+:::info
+💡 **Pro tip**: Did you know you can filter specific syscalls with `sudo ig run traceloop -c [container] --syscall-filters openat,write`? This helps reduce noise, and shows only failed system calls, helping you quickly identify issues when debugging file I/O issues.
+:::
+
+**Common filtering patterns:**
+```bash
+# Comprehensive file I/O testing
+--syscall-filters open,openat,close,read,write
+
+# Memory operations
+--syscall-filters mmap,munmap,brk,mprotect
+
+# Process management
+--syscall-filters fork,execve,exit,exit_group,wait4
+
+# Network debugging
+--syscall-filters socket,bind,listen,accept,connect,sendto,recvfrom
+```
+
+## Real-World Debugging Scenarios
```

Comment:
It could be named "Real-World Scenarios". Traceloop is not only about debugging but also for observability.

Reply:
Okay, I'll update this